crates/polars-mem-engine/src/executors/scan/parquet.rs

```diff
@@ -255,8 +259,7 @@ impl ParquetExec {
             let iter = readers_and_metadata.into_iter().enumerate().map(
                 |(i, (num_rows_this_file, reader))| {
                     let (remaining_rows_to_read, cumulative_read) = &rows_statistics[i];
-                    let hive_partitions = self
-                        .hive_parts
+                    let hive_partitions = hive_parts
```

Comment:
This is the fix for the corrupted reads in `read_async` - the index here is relative to the current batch.